### DISCUSSION TITLE
Is there some fundamental reason why debugger won't let us expand ref return properties?

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/csharplang#8007] [General] [answered]
I am unable to expand a property that's a `ref return`. If I have a `struct` that contains a pointer to another `struct` then that pointer field is expandable in debug, a property that returns that pointer is expandable in debug but a property that returns a `ref` created from the pointer is not expandable in debug. 

If I assign the property's returned `ref` to a `ref var` variable, that variable can be expanded fine.

Why is this? is there something fundamental that I'm overlooking?

I found this [issue report](https://developercommunity.visualstudio.com/t/ref-property-values-displayed-as-typename-with-no/521144) that appears to have just been ignored:

**Comment by HaloFour** [comment, marked as answer]
Maybe open in Roslyn?  This isn't a language or language design issue.

**Reply by The-Futurist** [reply]
OK, I'll do that, fair point.
